include/boost/redis/detail/multiplexer.hpp

```diff
@@ -116,8 +116,27 @@ class multiplexer {
       std::size_t read_size_;
    };
 
+   // This constructor makes testing easier
+   explicit multiplexer(bool conn_healthy = false);
+
+   // Handle connection health
+   void on_connection_up();  // Might be called in any state
+
+   void on_connection_down();  // Must be called once, with is_connection_healthy() == true
```

Comment:
This function is not needed since the connection is considered down when data is older than the ping interval specified in `config::health_check_interval`.